src-tests/test-sha256.hs

```diff
@@ -122,6 +126,12 @@ splitB l b
   where
     (b1, b2) = B.splitAt l b
 
+b16decode :: ByteString -> ByteString
+#if MIN_VERSION_base16_bytestring(1,0,0)
+b16decode = either error id . B16.decode
+#else
+b16decode = fst . B16.decode
+#endif
```

Comment:
What are the invariants on the inputs? Do we expect this to fail if there's any invalid hex, or do we want it to be lenient? `fst . B16.decode` would still product valid hex even in the presence of garbage, so you may want to leniently decode here. 

Otherwise, yes this is the right shape.

Reply:
AFAIU the new version of `b16decode` is stricter, because it checks that the full bytestring was decoded. Anyways, `b16decode` is used only in unit tests below (`rfc4231Tests` and `rfc5869Tests`) and inputs are always correctly encoded. 

`tasty-hunit` driver takes care to provide more context for `error`. E. g.,
```
      hmac:            FAIL
        Exception: invalid bytestring size
        CallStack (from HasCallStack):
          error, called at src-tests/test-sha256.hs:131:20 in main:Main
```

Reply:
Ah okay - good. Yeah, I was raising issue with the new version being stricter here if it touched any used code. No worries then!